benchmarks_test.go

```diff
@@ -0,0 +1,339 @@
+//   Copyright © 2015-2017 Ivan Kostko (github.com/ivan-kostko; github.com/gopot)
+
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+
+//       http://www.apache.org/licenses/LICENSE-2.0
+
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package errors_test
+
+import (
+	"fmt"
+	"testing"
+
+	e "errors"
+
+	"github.com/gopot/errors"
+)
+
+type CriticalError interface {
+	error
+	IsCritical() bool
+}
+
+func NewCriticalError(msg string, isCritical bool) *customErrorOneDetail {
```

Comment:
The naming is missleading in my opinion. `CriticalError` already implies that it is critical. So `isCritical` parameter is obsolete

Reply:
Good point. Will review naming in benchmarks.